### ISSUE TITLE
Should github.hosts be stored in local storage?

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode-pull-request-github#368]
Not sure it's useful to have in my settings.json, can this be hidden in local storage?

```
  "github.hosts": [
    {
      "host": "https://github.com",
      "username": "oauth",
      "token": "system"
    }
  ]
```

**Comment by shana** [comment]
This is in settings so that you can configure your own personal token for GitHub enterprise instances. The version of enterprise that you're running might not support the normal webflow signin process, in which case you'll need to manually configure a token in order to use it.

**Comment by Tyriar** [comment]
@shana why not store in local storage normally and allow overriding via settings.json if necessary then (or always fallback on the default)? This is really cluttering my settings file.

Also I notice now that `token` is no longer `system` but an actual token, is this sensitive as I would prefer not to have that in a file I manage/sync in the open. This could cause problems for the many users who use https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync

**Comment by shana** [comment]
@Tyriar When you say it's no longer `system`, do you mean you went through the sign in process and it added the entry in the settings file with `system`, and then sometime later it became the actual token?

Could you check your system credential manager and see if you have an entry there named `vscode-pull-request-github` with token in it, and whether it matches?

Having the configuration hidden away and able to be overridden by the setting file is a good idea, I'll see what I can do.

**Comment by queerviolet** [comment]
This should be fixed by #659
